components/eam/src/physics/cam/od_common.F90

```diff
@@ -392,14 +392,16 @@ function pblh_get_level_idx(height_array,pblheight)
   pblh_get_level_idx = -1
   found=.false.
   !get the pblh level index and return
-  do k = 1, pver
-    if((pblheight >= height_array(k+1).and.pblheight <height_array(k)))then
-      pblh_get_level_idx =  k+1
+  do k = 2, pver
+    if((pblheight >= height_array(k).and.pblheight <height_array(k-1)))then
+      pblh_get_level_idx =  k - 1
```

Comment:
@vanroekel , the change to `pblh_get_level_idx = k - 1` via commit https://github.com/E3SM-Project/E3SM/pull/7299/commits/b7289dae5abba168f905f9c5b63ffed65841ea10 moves it one level up. What is the consideration leading to this change?

Reply:
Thanks @wlin7 yes that's not right at all.  Just to confirm L397 should be `pblh_get_level_idx = k`?

Reply:
With line 397 changed back to `pblh_get_level_idx =  k`, a multi-year test with v3.LR confirms this PR is BFB with current master, as expected, when the new oro gravity drag scheme is turned on. That also means out-of-bound violation never occur even with the original code in the test (`v3.LR.F2010` for 6 years on chrysalis).